rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -118,10 +118,20 @@ void TimeSource::attachNode(
       return result;
     });
 
+  const rclcpp::QoS qos = (
+      rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(rmw_qos_profile_parameter_events))
+    );
+
+  rclcpp::SubscriptionOptions options;
+  options.content_filter_options.filter_expression
+    = std::string("node MATCH '") + node_base_->get_fully_qualified_name() + "'";;
```

Comment:
After setting content_filter_options for subscription while initialization, user can only use
https://github.com/iuhilnehc-ynos/rclcpp/blob/topic-content_filtered_topic/rclcpp/include/rclcpp/subscription_base.hpp#L282-L287
to update content filter expression.
(BTW: If not to set content_filter_options for subscription at the beginning, user also can update content filter expression by this interface `SubscriptionBase::set_cft_expression_parameters`)

> so this is opt-out, probably user should be able to configure this option?

What do you mean user should be able to configure this option?

User can't update the option in subscription because this option is constness.
https://github.com/iuhilnehc-ynos/rclcpp/blob/topic-content_filtered_topic/rclcpp/include/rclcpp/subscription.hpp#L341

Reply:
> What do you mean user should be able to configure this option?

yes, at the initialization time, for just once you can choose if they want to use CFT in builtin time source subscription.

at the moment, i was thinking that we change time source interface and user can choose if CFT is used in time source or not. e.g) NodeOptions::start_parameter_event_publisher for user application.

but i guess that is not needed now.

providing CFT option internally, it will be taken care of ros2 system, and on receiver it does check the event always no matter CFT is actually used on rmw implementation. it makes sense to me.